src/sqlancer/postgres/gen/PostgresMergeGenerator.java

```diff
@@ -0,0 +1,426 @@
+                if (i > 0) {
+                    sb.append(" AND ");
+                }
+                PostgresColumn column = columns.get(i);
+                sb.append("target.");
+                sb.append(column.getName());
+                sb.append(" = source.");
+                sb.append(column.getName());
+            }
+        } else {
+            PostgresColumn column = columns.get(0);
+            sb.append("target.");
+            sb.append(column.getName());
+            sb.append(" = source.");
+            sb.append(column.getName());
+        }
+
+        return sb.toString();
+    }
+
+    private static String generateUpdateSet(PostgresGlobalState globalState, PostgresTable targetTable,
+            List<PostgresColumn> sourceColumns) {
+        StringBuilder sb = new StringBuilder();
+
+        List<PostgresColumn> updateColumns = targetTable.getRandomNonEmptyColumnSubset();
+        for (int i = 0; i < updateColumns.size(); i++) {
+            if (i > 0) {
+                sb.append(", ");
+            }
+            PostgresColumn column = updateColumns.get(i);
+            sb.append(column.getName());
+            sb.append(" = ");
+
+            boolean sourceHasColumn = sourceColumns.stream().anyMatch(c -> c.getName().equals(column.getName()));
+
+            if (Randomly.getBoolean() && sourceHasColumn) {
+                sb.append("source.");
+                sb.append(column.getName());
+            } else {
+                PostgresExpressionGenerator gen = new PostgresExpressionGenerator(globalState)
+                        .setColumns(targetTable.getColumns());
+                PostgresExpression expression = gen.generateExpression(0, column.getType());
+                sb.append(PostgresVisitor.asString(expression));
+            }
+        }
+
+        return sb.toString();
+    }
+
+    private static String generateInsertValues(PostgresGlobalState globalState, List<PostgresColumn> columns,
+            List<PostgresColumn> sourceColumns) {
+        StringBuilder sb = new StringBuilder();
+
+        for (int i = 0; i < columns.size(); i++) {
+            if (i > 0) {
+                sb.append(", ");
+            }
+
+            PostgresColumn column = columns.get(i);
+
+            boolean sourceHasColumn = sourceColumns.stream().anyMatch(c -> c.getName().equals(column.getName()));
+
+            if (Randomly.getBoolean() && sourceHasColumn) {
+                sb.append("source.");
+                sb.append(column.getName());
+            } else {
+                PostgresExpressionGenerator gen = new PostgresExpressionGenerator(globalState).setColumns(columns);
+                PostgresExpression expression = gen.generateExpression(0, column.getType());
+                sb.append(PostgresVisitor.asString(expression));
+            }
+        }
+
+        return sb.toString();
+    }
+
+    private static String generateWhereCondition(PostgresTable targetTable) {
```

Comment:
I guess we can use the existing expression generator here? We can also add additional pseudo columns (like those prefixed by "target".